src/test/java/org/assertj/core/internal/objects/Objects_assertEqual_Test.java

```diff
@@ -80,10 +80,10 @@ public void should_fail_if_objects_are_not_equal_according_to_custom_comparison_
   }
 
   @Test
-  public void should_fail_with_my_exception_if_compared_with_null() {
+  public void should_fail_if_compared_with_null() {
     Throwable error = catchThrowable(() -> objects.assertEqual(someInfo(), new MyObject(), null));
 
-    assertThat(error).isInstanceOf(MyObject.NullEqualsException.class);
+    assertThat(error).isInstanceOf(AssertionError.class);
```

Comment:
This changed because `java.util.Objects#deepEquals` performs a null check first and then invokes `equals()`.